analyzers/src/SonarAnalyzer.CSharp/Rules/ApiControllersShouldDeriveFromController.cs

```diff
@@ -0,0 +1,101 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class ApiControllersShouldDeriveFromController : SonarDiagnosticAnalyzer
```

Comment:
Isn't the name of this analyzer the opposite of what the rule suggests?
Shouldn't it be `ApiControllersShouldDeriveFromControllerBASE`?

Reply:
Good catch!
I named it like that and then got confused as `Base` indicates a base class.
I renamed all the files to `ApiControllersShouldNotDeriveDirectlyFromController` so that it's clearer.

Reply:
I prefer to keep it as is.
It has happened a lot of times that I'm searching for a *Base in the solution to actually get a base class.
So IMO it will be confusing.
Happy to be contradicted.

Reply:
Sure, makes sense. :)